From f993697af81c37df9c55e0ebedeb1b8b880506ae Mon Sep 17 00:00:00 2001
From: Richard Leitner <richard.leitner@skidata.com>
Date: Tue, 5 May 2020 11:59:42 +0200
Subject: [PATCH] gdbm: fix link failure against gcc-10

Copied from gentoo's solution at https://bugs.gentoo.org/show_bug.cgi?id=705898
Original patch by Sergei Trofimovich <slyfox@gentoo.org>

Original description:

Before the change on gcc-10 link failed as:
```
  CCLD     gdbmtool
ld: ./libgdbmapp.a(parseopt.o):(.bss+0x8): multiple definition of `parseopt_program_args';
  gdbmtool.o:(.data.rel.local+0x260): first defined here
ld: ./libgdbmapp.a(parseopt.o):(.bss+0x10): multiple definition of `parseopt_program_doc';
  gdbmtool.o:(.data.rel.local+0x268): first defined here
```

gcc-10 will change the default from -fcommon to fno-common:
    https://gcc.gnu.org/PR85678.

The fix is to avoid multiple definition and rely on
declarations only.

Upstream-Status: Pending
Signed-off-by: Richard Leitner <richard.leitner@skidata.com>
---
 src/parseopt.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/parseopt.c b/src/parseopt.c
index 268e080..a4c8576 100644
--- a/src/parseopt.c
+++ b/src/parseopt.c
@@ -255,8 +255,6 @@ print_option_descr (const char *descr, size_t lmargin, size_t rmargin)
 }
 
 char *parseopt_program_name;
-char *parseopt_program_doc;
-char *parseopt_program_args;
 const char *program_bug_address = "<" PACKAGE_BUGREPORT ">";
 void (*parseopt_help_hook) (FILE *stream);
 
-- 
2.26.2

